2

```diff
@@ -3,11 +3,24 @@
 
 PATH=/usr/bin:/usr/sbin
 
+readproctitle='log: ...........................................................................................................................................................................................................................................................................................................................................................................................................'
+for arg in $(cat /proc/cmdline); do
+     case $arg in
+       'void.runit_readproctitle='* )
+           doreadproctitle="$(echo "${arg}" | cut -d'=' -f2)"
+           if [ "0" = "${doreadproctitle}" ]; then
+              echo 'runit readproctitle disabled: '"${arg}"' (via kernel cmdline)'
+              readproctitle=
+           fi
+       ;;
+     esac
+done
+
 runlevel=default
 for arg in $(cat /proc/cmdline); do
-    if [ -d /etc/runit/runsvdir/"$arg" ]; then
-        echo "Runlevel detected: '$arg' (via kernel cmdline)"
-        runlevel="$arg"
+    if [ -d /etc/runit/runsvdir/"${arg}" ]; then
+        echo 'runit runlevel detected: '"${arg}"' (via kernel cmdline)'
+        runlevel="${arg}"
```

Comment:
I'm not sure this is necessary... I also don't like how quoting is being done.

Reply:
Yes, you are right, that got in with my refactoring. 

First nitpick is, I guess, the long envvar form. I personally prefer "${xxx}" form as it removes 99.99% ambiguities and special cases, and if I remember correctly, it is still proper POSIX.

Despite it being longer to type, I believe it is more readable. This script is not written/edited often, and it does not impact performance. Also due to experiences from work$$$, when I find short forms (eg. "$xxx") I tend to rewrite them subconsciously and automatically to full forms. Seems like that happened here, sorry.

But I do realize now, it might look out of place WTR original void script style. Point taken.

Second nitpick is, I guess, "runit runlevel" string prefix. That was motivated by the fact runlevels are meaningless arbitrary numbers assigned ad-hoc. While other init systems may have many "runleves", runit has only three mandatory ones: 1(postboot/precruise) 2(cruise/main) and 3(shutdown). Other are user assigned. This edit was done, I think, to reduce confusion, that we are actually booting into "runit" runlevel, not "sys v" one (runit does not emulate "sys v" runlevels "systemd style"), but I digress.

Anyway these lines can be reverted back to originals, if it increases the chance of this commit getting in. Your call?